packages/@ourworldindata/utils/src/grapherConfigUtils.ts

```diff
@@ -0,0 +1,76 @@
+import { GrapherInterface } from "@ourworldindata/types"
+import {
+    isEqual,
+    mergeWith,
+    uniq,
+    omit,
+    excludeUndefined,
+    omitUndefinedValuesRecursive,
+    traverseObjects,
+} from "./Util"
+
+const KEYS_EXCLUDED_FROM_INHERITANCE = [
+    "$schema",
+    "id",
+    "slug",
+    "version",
+    "isPublished",
+]
+
+export function mergeGrapherConfigs(
+    ...grapherConfigs: GrapherInterface[]
+): GrapherInterface {
+    // warn if one of the configs is missing a schema version
+    const configsWithoutSchema = grapherConfigs.filter(
+        (c) => c["$schema"] === undefined
+    )
+    if (configsWithoutSchema.length > 0) {
+        const ids = configsWithoutSchema.map((c) => c.id)
+        console.warn(
+            `About to merge Grapher configs with missing schema information. Charts with missing schema information: ${ids}`
+        )
+    }
+
+    // abort if the grapher configs have different schema versions
+    const uniqueSchemas = uniq(
+        excludeUndefined(grapherConfigs.map((c) => c["$schema"]))
+    )
+    if (uniqueSchemas.length > 1) {
+        throw new Error(
+            `Can't merge Grapher configs with different schema versions: ${uniqueSchemas.join(
+                ", "
+            )}`
+        )
+    }
+
+    // keys that should not be inherited are removed from all but the last config
+    const cleanedConfigs = grapherConfigs.map((config, index) => {
+        if (index === grapherConfigs.length - 1) return config
+        return omit(config, KEYS_EXCLUDED_FROM_INHERITANCE)
+    })
+
+    return mergeWith(
+        {}, // mergeWith mutates the first argument
+        ...cleanedConfigs,
+        (_: unknown, childValue: unknown): any => {
+            // don't concat arrays, just use the last one
+            if (Array.isArray(childValue)) {
+                return childValue
+            }
+        }
+    )
+}
+
+export function diffGrapherConfigs(
+    config: GrapherInterface,
+    reference: GrapherInterface
+): GrapherInterface {
+    return omitUndefinedValuesRecursive(
+        traverseObjects(config, reference, (value, refValue, key) => {
+            if (KEYS_EXCLUDED_FROM_INHERITANCE.includes(key)) return value
```

Comment:
I wonder if it might bite us to use KEYS_EXCLUDED_FROM_INHERITANCE indiscriminately at all levels of recursion. If we were to add an `id` field inside the axis config in the future we might not bother to check here but introduce a hard to find bug. Similarly if we were to extend KEYS_EXCLUDED_FROM_INHERITANCE. 

Maybe it's better to just iterate over the KEYS_EXCLUDED_FROM_INHERITANCE once at the top of the object and delete each of those keys at the top level of (a copy of the) ref only? At the moment we only need to ignore top level values. If in the future we'd like to do arbitrary paths we could leverate the function `setValueRecursive` in `patchHelper.ts`

Reply:
Good point!